x-pack/plugins/endpoint/public/embeddables/resolver/models/process_event.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ProcessEvent } from '../types';
+
+export function isGraphableProcess(event: ProcessEvent) {
+  return eventType(event) === 'processCreated' || eventType(event) === 'processRan';
+}
+
+export function eventType(event: ProcessEvent) {
+  const {
+    data_buffer: { event_type_full: type, event_subtype_full: subType },
+  } = event;
+
+  if (type === 'process_event') {
+    if (subType === 'creation_event' || subType === 'fork_event' || subType === 'exec_event') {
+      return 'processCreated';
+    } else if (subType === 'already_running') {
+      return 'processRan';
+    } else if (subType === 'termination_event') {
+      return 'processTerminated';
+    } else {
+      return 'unknownProcessEvent';
+    }
+  } else if (type === 'alert_event') {
+    return 'processCausedAlert';
+  }
+  return 'unknownEvent';
+}
+
+export function uniquePidForProcess(event: ProcessEvent) {
+  return event.data_buffer.node_id;
```

Comment:
see comment about `source_id` stuff:

1. `v0` -- `endgame.unique_pid`
2. `v1` -- `endpoint.process.entity_id`

Reply:
I'm not totally sure how types will work with events, but I image having a single `Event` type which encompasses all of the events that we'll ever deal with. When we change that type, ts will help us fix all the code that references it. If we get sample data that has the right format, we'll change the type and all of the code then.